### ISSUE TITLE
Conker black line following your shadow

### ISSUE THREAD

**Issue by Tasosgemah** [gonetz/GLideN64#2112]
This line appears in the most recent version on GlideN64

![Untitled](https://user-images.githubusercontent.com/12124465/67762908-27498200-fa4f-11e9-8af5-02597cdb03a9.png)

It follows you around so i assume it's a shadow glitch.

**Comment by oddMLan** [comment]
Maybe related to #2097 ?

**Comment by Tasosgemah** [comment]
Not sure. But i get the same type of glitch in Jet Force Gemini as well.

https://i.postimg.cc/cZBZ7YrS/Jet-Force-Gemini-USA-191027-194631.png

**Comment by oddMLan** [comment]
It happens even at native resolution. I think this is definitely a regression caused the change to fix #2097, most likely.

**Comment by gonetz** [comment]
Do not use MSAA with CBFD.

**Comment by oddMLan** [comment]
![GLideN64_CONKER_BFD_000](https://user-images.githubusercontent.com/7278372/67741739-5d4b1f80-f9d7-11e9-98f1-7c2927274b35.png)
like I said it happens in native, that is without MSAA

**Comment by gonetz** [comment]
:(

**Comment by Tasosgemah** [comment]
I don't use MSAA or FXAA or any other similar option.

**Comment by DonelBueno** [comment]
It also causes similar issues in Mario Tennis:

![Project64_2019_10_30_14_48_15_715](https://user-images.githubusercontent.com/4239694/67884390-83caa100-fb24-11e9-8c4a-cfdb5b3a15de.png)

![Project64_2019_10_30_14_49_10_194](https://user-images.githubusercontent.com/4239694/67884401-86c59180-fb24-11e9-8fe2-f1be930bec1c.png)

**Comment by mudlord** [comment]
Noticed a similar thing with the intro on Paper Mario, on transitions.

**Comment by gonetz** [comment]
Duplicate of #519

**Comment by Tasosgemah** [comment]
But this wasn't an issue in recent versions, maybe it got fixed at some point and now it broke again?

**Comment by standard-two-simplex** [comment]
It seems it was this commit 0338f272 :(

**Comment by gonetz** [comment]
> Any chance that frame buffered textures are being shifted twice?

I suppose that the situation is opposite. This commit disables the shift for non-filtered textures. Filtered textures should work the same as before. So, rather the shift is missing somewhere.

**Comment by standard-two-simplex** [comment]
I just checked and conkers shadow seems to be a filtered texture. I also tried changing `if (uTextureFilterMode != 0) texCoordOut += vec2(0.5);` for `texCoordOut += vec2(0.5);` and the bug remains. Removing that lines solves the issue.

So I guess not all offsets were removed when `if (uTextureFilterMode != 0) texCoordOut += vec2(0.5);` was added. I tried looking for an unremoved `offset = 0.5`, but didn't find any.
